pkg/useragent/useragent.go

```diff
@@ -0,0 +1,61 @@
+// package useragent provides a consistent way to get a user agent for outbound http requests from Grafana Agent.
+// The default User-Agent is `GrafanaAgent/$VERSION($MODE)`
+// Where version is the build version of the agent and MODE is one of "static" or "flow".
+package useragent
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/grafana/agent/pkg/build"
+)
+
+const (
+	operatorEnv = "AGENT_OPERATOR"
+	modeEnv     = "AGENT_MODE"
+)
+
+// settable by tests
+var goos = runtime.GOOS
+
+func UserAgent() string {
+	parenthesis := ""
+	metadata := []string{}
+	if mode := getRunMode(); mode != "" {
+		metadata = append(metadata, mode)
+	}
+	metadata = append(metadata, goos)
+	if op := getOperator(); op != "" {
+		metadata = append(metadata, op)
+	}
+	if len(metadata) > 0 {
+		parenthesis = fmt.Sprintf(" (%s)", strings.Join(metadata, ";"))
```

Comment:
What do you think about using `strings.Join(metadata, "; ")` here (with the space after the semicolon)? That would help make the metadata slightly more human-readable:

```GrafanaAgent/v1.2.3 (static; linux; operator)```

vs 

```GrafanaAgent/v1.2.3 (static;linux;operator)```